src/components/launchermodel.cpp

```diff
@@ -726,7 +745,13 @@ LauncherItem *LauncherModel::addItemIfValid(const QString &path)
     LauncherItem *item = new LauncherItem(path, this);
 
     bool isValid = item->isValid();
-    bool shouldDisplay = item->shouldDisplay();
+    bool shouldDisplay = item->shouldDisplay() && _categories.isEmpty();
```

Comment:
Is it correct that if we have a matching category we will display, even if shouldDisplay() == false